# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/misc/directfb/i2c-hotfix.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

diff -urN -x '*~' DirectFB-0.9.18/gfxdrivers/matrox/matrox_maven.c DirectFB-0.9.18-fix/gfxdrivers/matrox/matrox_maven.c
--- DirectFB-0.9.18/gfxdrivers/matrox/matrox_maven.c	2003-06-02 17:28:24.000000000 +0200
+++ DirectFB-0.9.18-fix/gfxdrivers/matrox/matrox_maven.c	2003-10-02 07:53:46.000000000 +0200
@@ -21,14 +21,26 @@
    Boston, MA 02111-1307, USA.
 */
 
+/* TERRIBLE HACK */
+#define __KERNEL__
+
 #include <stdio.h>
 #include <unistd.h>
 #include <string.h>
 #include <ctype.h>
 #include <sys/types.h>
+#include <asm/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <sys/ioctl.h>
+
+// where the fsck does this get defined:
+typedef __s32 s32;
+typedef __u32 u32;
+typedef __u16 u16;
+typedef __u8 u8;
+
+#include <linux/i2c.h>
 #include <linux/i2c-dev.h>
 
 #include <misc/util.h>
